gm4_boat_chests/data/gm4_boat_chests/functions/merge.mcfunction

```diff
@@ -0,0 +1,12 @@
+
+advancement grant @a[distance=..5] only gm4:boat_chests
+
+tag @s add gm4_bc_old_boat
+function gm4_boat_chests:old_boat
+
+execute as @e[type=chest_minecart,distance=..3,sort=nearest,limit=1,tag=!gm4_bc_chest] run function gm4_boat_chests:old_minecart
+
+summon boat ~ ~ ~ {Type:"oak",Tags:["gm4_bc_boat","gm4_bc_new"],Passengers:[{id:"minecraft:chest_minecart",Tags:["gm4_bc_chest","gm4_bc_new"]}]}
+
+execute as @e[tag=gm4_bc_boat,tag=gm4_bc_new] run function gm4_boat_chests:new_boat
+execute as @e[tag=gm4_bc_chest,tag=gm4_bc_new] run function gm4_boat_chests:new_minecart
```

Comment:
```suggestion
summon boat ~ ~ ~ {Type:"oak",Tags:["gm4_bc_new"],Passengers:[{id:"minecraft:chest_minecart",Tags:["gm4_bc_new"]}]}
execute as @e[type=boat,tag=gm4_bc_new,limit=1] run function gm4_boat_chests:new_boat
execute as @e[type=chest_minecart,tag=gm4_bc_new,limit=1] run function gm4_boat_chests:new_minecart
```
Checking for `type` is better than checking for a `tag`, and `gm4_bc_chest`/`gm4_bc_boat` tags will be set in their respective functions, so the selectors and summon commands have been updated for efficiency.